examples/basic/main.c

```diff
@@ -0,0 +1,1021 @@
+      0x52, 0x53, 0x60, 0x6b, 0x89, 0xc9, 0xca, 0xed, 0x0a, 0x0c, 0x18, 0x33,
+      0x3c, 0x40, 0x42, 0x76, 0x7a, 0x7d, 0xb6, 0x10, 0x29, 0x3e, 0x42, 0x75,
+      0xa6, 0xbf, 0xcb, 0xd3, 0xdc, 0xe1, 0x00, 0x05, 0x08, 0x25, 0x67, 0x6c,
+      0x72, 0xde, 0x26, 0x72, 0x88, 0xbe, 0xfa, 0x05, 0x57, 0x77, 0x78, 0x83,
+      0xce, 0xd5, 0x44, 0x49, 0x8b, 0xab, 0xbc, 0xca, 0xd3, 0x1c, 0x34, 0x4a,
+      0xaa, 0xc7, 0xe5, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0c,
+      0x17, 0x22, 0x2a, 0x2f, 0x36, 0x3d, 0x43};
+#endif /* MLD_CONFIG_PARAMETER_SET == 87 */
+#endif /* !MLD_CONFIG_KEYGEN_PCT */
+  /* WARNING: Test-only
+   * Normally, you would want to seed a PRNG with trustworthy entropy here. */
+  randombytes_reset();
+
+  printf("ML-DSA-%d Basic Example\n", MLD_CONFIG_PARAMETER_SET);
+  printf("======================\n\n");
+
+  /* Initialize test message and context */
+  memcpy(m, test_msg, msglen);
+  memcpy(ctx, "test_context_123", ctxlen);
+
+  printf("Message: %.*s\n", (int)msglen, m);
+  printf("Context: %.*s\n\n", (int)ctxlen, ctx);
+
+  printf("Generating keypair ... ");
+
+  /* Alice generates a public/private key pair */
+  CHECK(crypto_sign_keypair(pk, sk) == 0);
+
+  printf("DONE\n");
+  printf("Signing message... ");
+
+  /* Alice signs the message */
+  CHECK(crypto_sign_signature(sig, &siglen, m, msglen, ctx, ctxlen, sk) == 0);
+
+  printf("DONE\n");
+  printf("Verifying signature... ");
+
+  /* Bob verifies Alice's signature */
+  CHECK(crypto_sign_verify(sig, siglen, m, msglen, ctx, ctxlen, pk) == 0);
+
+  printf("DONE\n");
+  printf("Creating signed message... ");
+
+  /* Alternative API: Create a signed message (signature + message combined) */
+  CHECK(crypto_sign(sm, &smlen, m, msglen, ctx, ctxlen, sk) == 0);
+
+  printf("DONE\n");
+  printf("Opening signed message... ");
+
+  /* Bob opens the signed message to recover the original message */
+  CHECK(crypto_sign_open(m2, &mlen, sm, smlen, ctx, ctxlen, pk) == 0);
+
+  printf("DONE\n");
+  printf("Compare messages... ");
+
+  /* Verify the recovered message matches the original */
+  CHECK(mlen == msglen);
+  CHECK(memcmp(m, m2, msglen) == 0);
+
+  printf("DONE\n\n");
+
+  printf("Results:\n");
+  printf("--------\n");
+  printf("Public key size:  %d bytes\n", CRYPTO_PUBLICKEYBYTES);
+  printf("Secret key size:  %d bytes\n", CRYPTO_SECRETKEYBYTES);
+  printf("Signature size:   %d bytes\n", CRYPTO_BYTES);
+  printf("Message length:   %zu bytes\n", msglen);
+  printf("Signature length: %zu bytes\n", siglen);
+  printf("Signed msg length: %zu bytes\n", smlen);
+
+  printf("\nComplete Signature (%zu bytes):\n", siglen);
+  printf("const uint8_t expected_signature[] = {\n    ");
+  {
+    size_t i;
+    for (i = 0; i < siglen; i++)
+    {
+      printf("0x%02x", sig[i]);
+      if (i < siglen - 1)
+      {
+        printf(", ");
+        if ((i + 1) % 12 == 0)
+        {
+          printf("\n    ");
+        }
+      }
+    }
+  }
+  printf("\n};\n");
```

Comment:
This pollutes the CI output somehow. Can we maybe only print the first 32 or 64 bytes or? Also no need to print this as a C array.